### DISCUSSION TITLE
Prevent header title changing to page title?

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#3813] [Q&A] [answered]
Hi! At present, the main title in the header changes to the page title on scroll.
So for example: https://squidfunk.github.io/mkdocs-material/getting-started/ "Material for MkDocs" changes to "Getting started" when you scroll the page.

Is it possible to disable this?

**Comment by squidfunk** [comment]
There's no feature flag, but you can override `header.html` and remove the following part:

https://github.com/squidfunk/mkdocs-material/blob/55a58da944e8f7521b2e81bd37b6eafd68cfa092/src/partials/header.html#L60-L67

**Reply by StarfallProjects** [reply]
Thanks! I ended up doing this to avoid the title vanishing:

```html
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            {{ config.site_name }}
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            {{ config.site_name }}
          </span>
        </div>
      </div>
    </div>
```

**Comment by mfncooper** [comment, marked as answer]
I know I'm late to the party, but ...

While the solution from @squidfunk does prevent the page title from replacing the site title, it causes the site title to go blank, rather than leaving it unchanged. For me, what @StarfallProjects used causes the site title to be replaced with the site title again but with a slightly different styling, so there's still a subtle change.

It turns out that simply removing the `data-md-component` attribute from this line does the right thing and prevents the site title from being replaced at all:

https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/header.html#L56

Hope that helps someone else digging for a solution.

**Reply by rodrigoalcarazdelaosa** [reply]
Is there any way to prevent this just on the homepage for example? I'm also using the `toc` extension (with `permalink: true`) and would like to disable it but just on the homepage. Thanks in advance!

**Reply by squidfunk** [reply]
Yes, you can override [`content.html`](https://github.com/squidfunk/mkdocs-material/blob/1b346eb57b2871eca8f5e78444cfc83d35ab2a2e/src/templates/partials/content.html) and remove the following lines:

https://github.com/squidfunk/mkdocs-material/blob/1b346eb57b2871eca8f5e78444cfc83d35ab2a2e/src/templates/partials/content.html#L31-L37

**Reply by rodrigoalcarazdelaosa** [reply]
Thanks, that helps me with the `<h1>` element, but if I add a `<h2>` heading, I still see the paragraph symbol. It'd be nice if we could control markdown extensions on a per page basis (setting them on the front matter), not sure how hard this could be. Thanks!

**Reply by squidfunk** [reply]
MkDocs is currently undergoing a major refactor, so it's a good idea to [ask the maintainers of MkDocs](https://github.com/mkdocs/mkdocs/issues) – maybe it's a good fit for the next version of MkDocs? Could be a sound use case, i.e., influencing Markdown rendering per page (or folder).

**Reply by rodrigoalcarazdelaosa** [reply]
For reference: https://github.com/mkdocs/mkdocs/discussions/3715
